src/templates/tag.js

```diff
@@ -111,7 +111,16 @@ module.exports = (page, metaIndex) => {
     ]
   };
 
-  return wrapper(page, metaIndex, html`
+  //because we're adding headers to the page, should update the headers list for ToC
+  const pageMetaForWrapper = {
+    ...page,
+    _headers: [
+      ...page._headers,
+      {title: "Tag structure", id: "tag-structure", level: 1}
+    ]
+  };
```

Comment:
Here's the pattern for augmenting the headers used in the ToC. When the source files are parsed and headers from markdown detected, those are put in `page._headers`. Instead of passing `page` to `wrapper()` directly, we'll insert an additional item into `_headers`. I'm doing this immutably with new objects/arrays and the spread operator because mutating method parameters is rude.